src/hooks/UrqlUseQuery.re

```diff
@@ -72,16 +72,28 @@ let useQuery =
   };
 
   let (stateJs, executeQueryJs) = useQueryJs(args);
+  let hasExecuted = React.useRef(false);
 
   let state =
-    React.useMemo2(
-      () => UrqlResponse.urqlResponseToReason(~response=stateJs, ~parse),
-      (stateJs, parse),
+    React.useMemo3(
+      () =>
+        UrqlResponse.urqlResponseToReason(
+          ~response=stateJs,
+          ~parse,
+          ~hasExecuted=
+            hasExecuted->React.Ref.current
+            || !pause->Belt.Option.getWithDefault(false),
```

Comment:
A little trickier with `useQuery`. We track `hasExecuted` as a ref and also look at the `pause` value. If `pause` is `false` we know the query has already executed, since queries run on mount in this case. If `pause` is `false` then we look at the `hasExecuted` ref.